machine_learning/Random Forest/RandomForest.py

```diff
@@ -0,0 +1,520 @@
+"""
+@author: Manish Kumar Chintha
+"""
+
+"""
+list of libraries imported for the program to work.
+"""
+import math
+import numpy as np
+import csv
+from collections import defaultdict
+import operator
+import random
+
+"""
+Class for DecisionTree classifier and all of it's related model.
+I have called this class in RandomForest Class to build multiple tree models
+and saved the objects and models in RandomForest class to use for classification
+
+Building a Decision Tree and testing it on  training data.
+I've separated the Decision Tree classifier and related function,
+so that I can even test it independently.
+"""
+
+
+class DecisionTree(object):
+    """
+    class attributes; classIndex define the target class index in the data. In
+    our case target class index is always the last index of the data. training
+    sampels is the training data provided to Decision Tree Class every time
+
+    We're getting random bootstrap sample data with replacement every time
+    for each decision tree model built:
+    """
+
+    classIndex = 0
+    trainingSamples = []
+
+    """
+    init function to initialize the classIndex and trainingSamples
+    """
+
+    def __init__(self, classIndex, trainingSamples):
+        self.classIndex = classIndex
+        self.trainingSamples = trainingSamples
```

Comment:
Dataclasses make this easier...  All methods will need type hints and doctests.
```suggestion
    class_index: int
    training_samples: list[]
```